kyaml/kio/byteio_readwriter_test.go

```diff
@@ -324,3 +324,271 @@ functionConfig:
 		})
 	}
 }
+
+func TestByteReadWriter_RetainSeqIndent(t *testing.T) {
+	type testCase struct {
+		name           string
+		err            string
+		input          string
+		expectedOutput string
+		instance       kio.ByteReadWriter
+	}
+
+	testCases := []testCase{
+		{
+			name: "round_trip with 2 space seq indent",
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+spec:
+  - foo
+  - bar
+---
+apiVersion: v1
+kind: Service
+spec:
+  - foo
+  - bar
+`,
+			expectedOutput: `
+apiVersion: apps/v1
+kind: Deployment
+spec:
+  - foo
+  - bar
+---
+apiVersion: v1
+kind: Service
+spec:
+  - foo
+  - bar
+`,
+		},
+		{
+			name: "round_trip with 0 space seq indent",
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+spec:
+- foo
+- bar
+---
+apiVersion: v1
+kind: Service
+spec:
+- foo
+- bar
+`,
+			expectedOutput: `
+apiVersion: apps/v1
+kind: Deployment
+spec:
+- foo
+- bar
+---
+apiVersion: v1
+kind: Service
+spec:
+- foo
+- bar
+`,
+		},
+		{
+			name: "round_trip with different indentations",
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+spec:
+  - foo
+  - bar
+  - baz
+---
+apiVersion: v1
+kind: Service
+spec:
+- foo
+- bar
+`,
+			expectedOutput: `
+apiVersion: apps/v1
+kind: Deployment
+spec:
+  - foo
+  - bar
+  - baz
+---
+apiVersion: v1
+kind: Service
+spec:
+- foo
+- bar
+`,
+		},
+		{
+			name: "round_trip with mixed indentations in same resource, wide wins",
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+spec:
+  - foo
+  - bar
+  - baz
+env:
+- foo
+- bar
+`,
+			expectedOutput: `
+apiVersion: apps/v1
+kind: Deployment
+spec:
+  - foo
+  - bar
+  - baz
+env:
+  - foo
+  - bar
+`,
+		},
+		{
+			name: "round_trip with mixed indentations in same resource, compact wins",
```

Comment:
s/,compact wins/with compact indentation in first sequence

Reply:
Done